src/vmm/src/device_manager/legacy.rs

```diff
@@ -169,6 +170,73 @@ impl PortIODeviceManager {
     }
 }
 
+impl Aml for PortIODeviceManager {
+    fn append_aml_bytes(&self, v: &mut Vec<u8>) {
+        // Set up COM devices
+        let gsi = [
+            Self::COM_EVT_1_3_GSI,
+            Self::COM_EVT_2_4_GSI,
+            Self::COM_EVT_1_3_GSI,
+            Self::COM_EVT_2_4_GSI,
+        ];
+        for com in 0u8..4 {
+            // COM1
+            aml::Device::new(
+                format!("_SB_.COM{}", com + 1).as_str().into(),
+                vec![
```

Comment:
Do we need to allocate vectors here? I seems the data is always the same. Overall can we make acpi tables no to do any unnecessary allocations?

Reply:
Unfortunately, at the moment yes. This is how the `aml` module handles collections of other objects. Most likely, we can improve this, but I don't think this is a priority right now. My measurements say that building everything need for ACPI takes a few tens of microseconds.